.github/workflows/release.yaml

```diff
@@ -0,0 +1,35 @@
+name: release
+on:
+  push:
+    tags:
+    - '*'
+
+jobs:
+  release:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    if: github.repository == 'pytest-dev/pytest-timeout'
+
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.7
+
+      - name: Install dependencies
+        run: |
+          python -m pip install -U pip
+          python -m pip install -U build pytest>=5.0.0
+
+      - name: Build package
+        run: |
+          python -m build
+          python -m twine check dist/*
+
+      - name: Publish package distributions to PyPI
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```

Comment:
I believe this `if` is not necessary, given the trigger is already push on tags?

Reply:
@nicoddemus Should I use this even instead?
```yaml
on:
  release:
    types: [published]
```
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

Reply:
Yeah, I think using `on release` and ditching the if statement is the best and simplest approach.
https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

Reply:
Never used this option, but seems reasonable to require a release rather than a direct tag.

Other repos might do the other way around: push a tag, and one of the things done at the end of the workflow after uploading the package to PyPI is creating a release.